pkg/provider/aws/secretsmanager/secretsmanager.go

```diff
@@ -71,6 +70,7 @@ type SecretsManager struct {
 	cache        map[string]*awssm.GetSecretValueOutput
 	config       *esv1.SecretsManager
 	prefix       string
+	newUUID      func() string
```

Comment:
This is never set (other than in one test).  Would it be a better idea to set it to `uuid.NewString()` in the `New` func (~ line 109)?  Then, you don't need to test for it in `putSecretValueWithContext` (line 543.)

Wether is makes sense to take it as a (nil-able) argument in `New`, I don't know.

Reply:
Yep, what Jakob said. It's defensive programming basically. If anyone ever uses the struct uninitialized, they will get a nil. So I'd rather just check in that one place where it's actually used.